src/Quadrature/quadrature.jl

```diff
@@ -15,8 +15,8 @@ using Base.Cartesian: @nloops, @ntuple, @nexprs
     QuadratureRule{shape}(weights::AbstractVector{T}, points::AbstractVector{Vec{rdim, T}})
 
 Create a `QuadratureRule` used for integration on the refshape `shape` (of type [`AbstractRefShape`](@ref)).
-`order` is the order of the quadrature rule.
-`quad_rule_type` is an optional argument determining the type of quadrature rule,
+`order` is related to the number of quadrature points per direction (i.e. a `QuadratureRule{RefQuadrilateral}(2)` 
+uses 2 quadrature points per direction). `quad_rule_type` is an optional argument determining the type of quadrature rule,
```

Comment:
This is only true for hypercubes. For simplexes the concept of "each direction" is a bit harder to describe. I don't think we are fully consistent with what is meant with order, see also #1022 for some discussions about the options. For now, would something like this explain good enough? 
```suggestion
`order` is related to the number of quadrature points, but is dependent on the reference shape. 
`quad_rule_type` is an optional argument determining the type of quadrature rule,
```

Reply:
I discussed that with Phil: We thought the general "is related to" would be enough to make clear that this is not always exactly true
Now, I see that this is still not perfectly clear.
What you suggest could be misread as "The order depends on the reference shape." But this is not what you mean, right?

Reply:
Ohh, good catch here! Not sure how we missed that inconsistency earlier. We should definitely clarify this.

When writing the docs I found it difficult to make a balance between 1) being precise and 2) being still useful for users which are not fully aware (and interested) of all details. So for everything but hypercubes we refer with order to the actual integration order of the rule w.r.t. to the function system it integrates. Note that the function system varies between rules, see for example the [Gauss-Laguerre rule](https://en.wikipedia.org/wiki/Gauss%E2%80%93Laguerre_quadrature) for an example of a fancy system. So, if you have any idea how to communicate what "order" means to users, I am open for any suggestion.

Reply:
Good point! 
Something like this, and if added in topics or somewhere else, we should document the rules somewhere and point to that location. Probably nice not having it in the docstring directly, as this would get too verbose. 

```suggestion
`order` affects the number of quadrature points, and thus the precision of the integration rule. The exact relationship between `order` and the number of quadrature points depends on the `quad_rule_type` and `shape`.
`quad_rule_type` is an optional argument determining the type of quadrature rule,
```

Reply:
I think this is nice for the docstring :+1: 
Maybe we could add a more detailed description (hypercubes vs. other shapes) somewhere in the documentation?
Suggestions where to add it:

1) [at the beginning of "Topic guides/Reference shapes"](https://github.com/Ferrite-FEM/Ferrite.jl/blob/3b081bea81cb718cf33b004ea16f90fdbb8cc5f9/docs/src/topics/reference_shapes.md?plain=1#L14)

2) [at the beginning of "Topic guides/Reference shapes"](https://github.com/Ferrite-FEM/Ferrite.jl/blob/3b081bea81cb718cf33b004ea16f90fdbb8cc5f9/docs/src/topics/reference_shapes.md?plain=1#L84)

3) [in the tutorial "Heat equation"](https://github.com/Ferrite-FEM/Ferrite.jl/blob/3b081bea81cb718cf33b004ea16f90fdbb8cc5f9/docs/src/literate-tutorials/heat_equation.jl#L56)